extra/mariabackup/backup_mysql.cc

```diff
@@ -1432,6 +1435,22 @@ write_xtrabackup_info(MYSQL *connection, const char * filename, bool history)
 		xb_stream_name[xtrabackup_stream_fmt], /* format */
 		xtrabackup_compress ? "compressed" : "N"); /* compressed */
 
+	if (stream) {
+		backup_file_printf(filename, "%s", buf);
+	} else {
+		fp = fopen(filename, "w");
+		if (!fp) {
+			msg("Error: cannot open %s", filename);
+			result = false;
+			goto cleanup;
+		}
+		if (fwrite(buf, buf_len, 1, fp) < 1) {
+			fclose(fp);
```

Comment:
No need to close `fp` here, it will be closed at the end (or you can bring the fclose here, I'm ok with both). An error message here would be helpful in case of a short write.